samples/client/petstore/java-micronaut-client/src/main/java/org/openapitools/model/Animal.java

```diff
@@ -148,7 +144,9 @@ private String toIndentedString(Object o) {
      * @param <T> the return type of the visitor
      * @return the result from the visitor
      */
-    public abstract <T> T accept(Visitor<T> visitor);
+    public <T> T accept(Visitor<T> visitor) {
```

Comment:
I don't think this is correct. Shouldn't the pojo be abstract (or even an interface) when it has a discriminator? Or is it possible to (de)serialize an instance of `Animal` which isn't a BigCat, Cat or Dog?

Reply:
@auke- thanks for your comment.
It seems to be standard for generated models to not be `abstract`. It seems like it is possible to deserialize the parent and I think there are cases when it makes sense. For example, with such configuration:
```yaml
components:
  schemas:
    BookAvailability: 
      type: string
      enum: ["available", "not available", "reserved"]
    Book:
      title: Book Info
      description: Object containg all the info about a book
      type: object
      discriminator:
        propertyName: bookType
        mapping:
          book: "#/components/schemas/Book"
          encyclopedia: "#/components/schemas/Encyclopedia"
      properties:
        name: {type: string}
        availability: {$ref: "#/components/schemas/BookAvailability"}
        author: {type: string, minLength: 3}
        ISBN: {type: string, pattern: "[0-9]{13}"}
      required: ["name", "availability"]
    Encyclopedia:
      title: Encyclopedia Info
      type: object
      allOf: [{$ref: "#/components/schemas/Book"}]
      properties:
        topic: {type: string, enum: ['biology', 'math', 'chemistry', 'physics']}

```
As you can see a `Book` instance can exist by itself and even has an own mapping. 

In case of pet store example, it might also be strange if `Animal` was `abstract`, while `Cat` wasn't, since `Cat` has the `BigCat` subclass and is not a node in the inheritance tree:
```
Animal (?abstract)
├── Dog
└── Cat (?abstract)
    └── BigCat
```

However, I do agree that in some cases it would make more sense to have an abstract object. For example, if the parent class does not have a mapping and specifies that the `oneOf` property. For example:
```yaml
components:
  schemas:
    Animal:
      type: object
      discriminator:
        propertyName: animalType
        mapping:
          cat: "#/components/schemas/Cat"
          dog: "#/components/schemas/Dog"
      properties:
        name: {type: string, description: The name of the animal}
      oneOf: [{$ref: "#/components/schemas/Cat"}, {$ref: "#/components/schemas/Dog"}]
    Cat:
      type: object
    Dog:
      type: object
```
In this case the definition clearly says that `Animal` is "one of" cat or dog, and cannot exist as an instance itself.

So, to me it seems like in some cases the parent class should be `abstract`, but not always. I would keep the current implementation for now, (since it is also standard for other libraries in generator), but consider making some improvements to the handling of `discriminator` in another PR.

@auke- what do you think? PS you can also contact me in the openapi slack: https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

@wing328 do you think I can open an issue to discuss the discriminator implementation in `Java` in general, or is there a better place to do the discussion?

Reply:
@andriy-dmytruk something like this https://github.com/OpenAPITools/openapi-generator/pull/12361 ?